pkg/daemon/file_writers.go

```diff
@@ -45,12 +46,42 @@ func noOrigFileStampName(fpath string) string {
 	return filepath.Join(noOrigParentDir(), fpath+".mcdnoorig")
 }
 
-func createOrigFile(fromPath, fpath string) error {
-	if _, err := os.Stat(noOrigFileStampName(fpath)); err == nil {
-		// we already created the no orig file for this default file
-		return nil
+func createOrigFile(fromPath, fpath string, isCoreOSVariant, forceCreate bool) error {
+	orig := false
+
+	// https://issues.redhat.com/browse/OCPBUGS-11437
+	// MCO keeps the pull secret to .orig file once it replaced
+	// Adapt a check used in function deleteStaleData for backwards compatibility: basically if the file doesn't
+	// exist in /usr/etc (on FCOS/RHCOS) and no rpm is claiming it, we assume the orig file doesn't need to be created
+	if _, err := exec.Command("rpm", "-qf", fpath).CombinedOutput(); err == nil {
+		// File is owned by an rpm
+		orig = true
+	} else if strings.HasPrefix(fpath, "/etc") && isCoreOSVariant {
+		if _, err := os.Stat(withUsrPath(fpath)); err != nil {
+			if !os.IsNotExist(err) {
+				return err
+			}
+		} else {
+			orig = true
+		}
+	} else if forceCreate {
```

Comment:
nit: since forceCreate always makes `orig` true, it might be better to have this condition check first, to save some extra operations